### PR TITLE
Config names no longer overlap

### PR DESCRIPTION
**Pull request by Vbeany**
Known bugs: Changing name of last config gives Request NaN for next config. Made separate issue #573

### CODE REVIEW

**Comment by amgrubb** [comment]
I changed the base to `sep21-bugs`.

**Review by i-m-foster — APPROVED** [review]
It looks good to me! I would add the bug number it fixes in a comment so it's easy to see and maybe also add the bug number to the linked issues section on the side.

File: leaf-ui/rappid-extensions/ConfigInspector.js [hunk 1]
```diff
@@ -347,7 +347,13 @@ var ConfigInspector = Backbone.View.extend({
 
     /** Create and add a new config model to the collection */
     addNewConfig: function () {
-        var configModel = new ConfigBBM({ name: "Request " + (this.collection.length + 1), results: new ResultCollection([]) })
+        var configModel; 
+        if (this.collection.length < 1){
+                configModel = new ConfigBBM({ name: "Result " + 1, results: new ResultCollection([]) });
```
Comment (krhablutzel) [inline leaf-ui/rappid-extensions/ConfigInspector.js]: We want to call these "Request", not "Result"

File: leaf-ui/rappid-extensions/ConfigInspector.js [hunk 2]
```diff
@@ -347,7 +347,13 @@ var ConfigInspector = Backbone.View.extend({
...
+        } else { 
+            var nameIndex = parseInt(this.collection.at(this.collection.length - 1 ).get('name').split(' ').pop()) + 1;
```
Comment (krhablutzel) [inline leaf-ui/rappid-extensions/ConfigInspector.js:354]: When a request is renamed to something not containing a number, this replaces the last word in the name with `NaN`. This also returns odd results when a Request is named just a number or has an empty name. (Also, less importantly, if we rename a Request to a different number, we start counting from that number even if that isn't the total number of Requests ever generated - not sure whether we like this behavior.)

Screen Shot 2021-09-18 at 13 34 47

**Review by anishaj149 — CHANGES_REQUESTED** [review]
(Worked with Kathleen so her comment contains necessary changes.) :)

**Review by krhablutzel — CHANGES_REQUESTED** [review]
This works well when the user has not renamed Requests (other than the Result/Request typo). However, it generates some odd behavior when the user has renamed the Request (see comments below).

**Review by krhablutzel — APPROVED** [review]
Approving this since the noted changes are covered in separate issue #573

**Pull request merged**